src/app/shared/components/search-input/search-input.component.ts

```diff
@@ -37,13 +45,28 @@ export class SearchInputComponent implements OnInit, OnDestroy {
     this.destroy.complete();
   }
 
+  public writeValue(search: string): void {
+    this.searchText.setValue(search);
+  }
+
+  public onChange = (search: string) => {};
+  public onTouched = () => {};
+
+  public registerOnChange(fn: (search: string) => void): void {
+    this.onChange = fn;
+  }
+
+  public registerOnTouched(fn: () => void): void {
+    this.onTouched = fn;
+  }
+
   private subscribeToSearch(): void {
     this.searchText.valueChanges.pipe(
-      debounceTime(1000),
+      debounceTime(300),
       distinctUntilChanged(),
       takeUntil(this.destroy)
     ).subscribe((text) => {
-      this.store.dispatch(fromRoot.Go({ path: ['/cards'], query: { search: text } }));
+      this.onChange(text);
```

Comment:
Now we'll simply output the value and let whatever is housing this component control what to do with the search when it outputs.